checkstyle/src/main.rs

```diff
@@ -28,78 +25,76 @@ fn read_file_content(file_path: &Path) -> String {
+        return Some(false);
+    }
+    let together = file_lines.iter().zip(regex_lines.iter());
+    for (file_line, regex_line) in together {
+        let re = Regex::new(regex_line).ok()?;
+        if !re.is_match(file_line) {
+            return Some(false);
+        }
+    }
+    Some(true)
+}
+
+fn check_matching(file: &str, regex: &str) -> bool {
     let file_lines: Vec<&str> = file.lines().collect();
     let regex_lines: Vec<&str> = regex.lines().collect();
-    let mut file_id = 0;
-    let mut regex_id = 0;
-    while file_id < file_lines.len() && regex_id < regex_lines.len() {
-        let regex_line = regex_lines[regex_id];
-        let file_line = file_lines[file_id];
-        let re = Regex::new(regex_line).ok()?;
-        if re.is_match(file_line) {
-            file_id += 1;
-            regex_id += 1;
-        } else {
-            file_id += 1;
+    for i in 0..file_lines.len() {
+        let suffix = &file_lines[i..];
+        if check_direct_matching(suffix.to_vec(), regex_lines.to_vec()).unwrap_or(false) {
+            return true;
         }
     }
-    Some(regex_id == regex_lines.len())
+    return false;
 }
 
-fn list_files_in_folder(folder_path: &Path) -> Result<Vec<String>, std::io::Error> {
+fn list_files_in_folder(folder_path: &Path) -> Result<Vec<PathBuf>, std::io::Error> {
     let mut file_paths = Vec::new();
 
     for entry in fs::read_dir(folder_path)? {
         let entry = entry?;
         let path = entry.path();
 
         if path.is_file() {
-            if let Some(file_name) = path.file_name() {
-                if let Some(file_name_str) = file_name.to_str() {
-                    file_paths.push(file_name_str.to_string());
-                }
+            file_paths.push(path.clone());
+        } else if path.is_dir() {
+            if let Ok(sub_paths) = list_files_in_folder(&path) {
+                file_paths.extend(sub_paths);
             }
         }
     }
 
     Ok(file_paths)
 }
 
-fn all_files(folder_path: &Path) -> Vec<String> {
+fn all_files(folder_path: &Path) -> Vec<PathBuf> {
     list_files_in_folder(folder_path).unwrap_or_default()
 }
 
-fn separate_regex_matching_files(
-    regex_file_path: &Path,
-    target_path: &Path,
-) -> (Vec<String>, Vec<String>) {
+fn separate_regex_matching_files(regex_file_path: &Path, target_path: &Path) -> Vec<PathBuf> {
     let target_file_paths = all_files(target_path);
     let regex = read_file_content(regex_file_path);
-
-    let (matching_files, nonmatching_files): (Vec<String>, Vec<String>) =
+    let (_matching_files, nonmatching_files) =
         target_file_paths.into_iter().partition(|target_file_name| {
             let target_file = read_file_content(&target_file_name.as_ref());
-            let is_match_option = check_matching(&target_file, &regex);
-            is_match_option.unwrap_or(false)
+            check_matching(&target_file, &regex)
         });
 
-    (matching_files, nonmatching_files)
+    nonmatching_files
```

Comment:
The partition is a neat function, but now that we're not paying attention to the matching part we can just turn this into a normal filter.